packages/components/src/Menu/Menu.tsx

```diff
@@ -208,15 +214,30 @@ export interface ActionProps {
    * Callback when an action gets clicked
    */
   onClick?(event: React.MouseEvent<HTMLButtonElement>): void;
+
+  /**
+   * Focuses explicitly if set to true
+   */
+  shouldFocus?: boolean;
 }
 
-function Action({ label, icon, onClick }: ActionProps) {
+function Action({ label, icon, onClick, shouldFocus = false }: ActionProps) {
+  // eslint-disable-next-line no-null/no-null
+  const actionButtonRef = useRef<HTMLButtonElement | null>(null);
+
+  useEffect(() => {
+    if (actionButtonRef.current && shouldFocus) {
+      actionButtonRef.current.focus();
+    }
+  }, [shouldFocus]);
```

Comment:
@chris-at-jobber I hope you might know better, but I noticed in the mui menu component that the first item is not focused when using the mouse, but focused when using keyboard. In this implementation, it's focusing on both onClick and when using keyboard. Should that be the case?

Reply:
I am actually working on a11y this close for the first time. Just going by the video. Somehow, I am not able to replicate the interaction with just tab 😕.
Is it possible for you to tell me how you interacted? I am using tab after a click but it is closing the menu for me. Thanks!

Reply:
> @chris-at-jobber I hope you might know better, but I noticed in the mui menu component that the first item is not focused when using the mouse, but focused when using keyboard. 

I am sorry, this wasn't meant for me but for @chris-at-jobber . My bad.

Reply:
@joelwarrington-jobber that's a great catch! The first item should only receive focus on the keyboard interaction; the mouseclick should retain focus on the clicked trigger element. Here's a video showing both interactions:

https://user-images.githubusercontent.com/39704901/135682590-8e1e2886-c7bf-4906-9b2d-213024e9a03b.mov
 
Note that on the first interaction (keyboard) we have the desired interaction. I move focus to the menu trigger, action the menu "trigger" button with `space`/`return`, focus moves to the first menu item, and I can hit `return` or `space` to action the menu item.

On the second interaction (mouse click) I'm able to hit `return` to action the first menu option, but I should not be able to. See this [demo from w3 org](https://www.w3.org/WAI/tutorials/menus/application-menus-code/) on mouse-click events for reference.

Tested in Menu's usage within the [Page component.](https://deploy-preview-706--jobber-atlantis.netlify.app/components/page#page-with-actions)

Reply:
Thanks for the details here @chris-at-jobber. I am still trying to understand the mouse interaction. From the `w3` org, what I could see is I am able to hit return and see `100%` when I 
1. Click on `View` 
2. Press `return` 

![image](https://user-images.githubusercontent.com/15359575/135685060-bbc2311c-d0a3-4965-8303-c3b2130173e9.png)

And similar is what I can see from the current implementation as well. 

In the `w3`, if you:
1. Click on `View` 
2. Move the cursor away, it comes with a focussed white background and I can click return to alert.

![image](https://user-images.githubusercontent.com/15359575/135685313-72719e51-7996-4c2d-81de-60b6a4c7cfc2.png)


In both cases - mouse and keyboard - hitting `return` is showing the alert with focus. Any idea on what I might be missing? Apologies for too many follow-ups.

Reply:
You know what, you're right! The visual presentation of their demo made it less apparent that the focus was in fact set, but the same behaviour is there. No changes required on this topic 👍